### PR TITLE
Add lint staged

### PR DESCRIPTION
**Pull request by anusha5695**
Add lint staged:

Issue: https://github.com/CatalysmsServerManager/7-days-to-die-server-manager/issues/435

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -146,7 +148,10 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "npm run lint"
+      "pre-commit": "lint-staged"
     }
+  },
+  "lint-staged": {
+    "*.js": "npm run lint"
```
Comment (halkeye) [inline package.json:155]: lets change this to npx eslint since run lint does everything

File: package.json [hunk 2]
```diff
@@ -83,14 +83,16 @@
...
+    "husky": "^5.0.0-beta.0",
+    "lint-staged": "^10.4.2",
     "mocha": "^8.1.0",
     "mocha-junit-reporter": "^2.0.0",
     "mocha-lcov-reporter": "^1.3.0",
     "mock-req-res": "^1.2.0",
     "mockdate": "^3.0.2",
     "nodemon": "^2.0.3",
     "nyc": "^15.0.1",
+    "prettier": "^2.1.2",
```
Comment (halkeye) [inline package.json:95]: why was prettier added?

File: package.json [hunk 3]
```diff
@@ -83,14 +83,16 @@
     "eslint": "^7.4.0",
     "faker": "^4.1.0",
     "grunt-bower-task": "^0.5.0",
-    "husky": "^4.3.0",
+    "husky": "^5.0.0-beta.0",
```
Comment (halkeye) [inline package.json:86]: I use husky 5 myself (as a supporter), but I don't think its really ready to use fore everyone, is there a reason you upgraded?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "csmm",
-  "version": "1.19.0",
+  "version": "1.19.1",
```
Comment (halkeye) [inline package.json:3]: this get bumped on release, so lets leave it as 0 for now.
```suggestion
  "version": "1.19.0",
```

File: package.json (hunk 2 above)
Comment (halkeye) [reply package.json:95]:
```suggestion
```

File: package.json (hunk 3 above)
Comment (halkeye) [reply package.json:86]:
```suggestion
     "husky": "^4.3.0",
```

File: package.json (hunk 1 above)
Comment (halkeye) [reply package.json:155]:
```suggestion
    "*.js": "npx eslint"
```

File: package.json (hunk 3 above)
Comment (anusha5695) [reply package.json:86]: Oh sorry I was trying out on local machine for w/ versions of husky. sorry would get this removed asap

File: package.json (hunk 4 above)
Comment (anusha5695) [reply package.json:3]: sure

File: package.json (hunk 2 above)
Comment (anusha5695) [reply package.json:95]: This as well, I was experimenting this before committing. Will get this rectified. :) Thanks for the patience

**Comment by anusha5695** [comment]
@halkeye have fixed all the feedback

**Review by niekcandaele — APPROVED** [review]

**Comment by niekcandaele** [comment]
LGTM, thanks for your contribution!

@halkeye merge this when you've reviewed the latest changes?

**Review by halkeye — APPROVED** [review]
Lgtm

**Pull request merged**